File: Dynamic Programming/Coin_Combinations_I.py
# Problem Link :- https://cses.fi/problemset/task/1635

n,x = map(int,raw_input().split())
arr = list(map(int,raw_input().split()))
arr.sort()
dp = [0 for i in range(x+1)]
dp[0] = 1
mod = 1000000007
for j in range(1,x+1):
    for i in range(n):
        if j >= arr[i]:
            dp[j] += (dp[j-arr[i]])%mod
            dp[j] %= mod
print(dp[x])

File: Dynamic Programming/Coin_Combinations_I.cpp
// Problem Link :- https://cses.fi/problemset/task/1635

#include<iostream>
#include<bits/stdc++.h>
using namespace std;
#define ll long long int
const int MOD = 1000000007;

int main(){
    ll n,x;
    cin >> n >> x;
    vector<ll> arr(n);
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }
    sort(arr.begin(),arr.end());
    vector<ll> dp(x+1);
    dp[0] = 1;
    for(int j=1;j<=x;j++){
        for(int i=0;i<n;i++){
            if (j >= arr[i]){
                dp[j] += (dp[j-arr[i]])%MOD;
                dp[j] %= MOD;
            }    
        }
    }
    cout << dp[x] <<'\n';
    return 0;
}